### DISCUSSION TITLE
Question about plotting IPDF with the coloring based on the magnitude of  a specific property for the corresponding orientation

### DISCUSSION THREAD

**Discussion by vivek9626** [mtex-toolbox/mtex#1388] [Ask Anything] [answered]
Hello all,

I want to make distributions similar to the below figure (Image Referred from https://www.sciencedirect.com/science/article/pii/S1359645416307972).

![image](https://user-images.githubusercontent.com/74325642/159714122-fa0cf391-bfca-4d50-ac97-7b80f5a270cb.png)

For now, I can make population distributions (figure 9b and 9e) using mtex using the below code:

hkldouble = double(v.hkl);
uvwdouble = double(v.uvw);
ori2 = orientation.byMiller(hkldouble,uvwdouble,cs);
odf = calcDensity(ori2);
figure
plotIPDF(odf,[zvector])

However, I cannot do figure 9a 9c 9d 9f. I have all the curvature values stored in an array for each of the orientations of interest. I am not sure how to change the coloring based on a specific value for a grid in the distribution instead of having the coloring based on the population. I am not sure if MTEX has the ability to do that using the predefined functions. If MTEX predefined functions cannot do the job is it possible for anyone to guide me on how I can create my own little function to do the distribution plots based on a specific property for different orientations.

Thank you for being so helpful.

Regards,
Vivek

**Comment by kilir** [comment]
Hi,
actually, if you apply the orientations as the first and the data as the second argument to plotIPDF, e.g. 'contourf' should interpolate the data. Also have a look at `Miller/interp`.

Hope that helps.

Cheers,
Rüdiger

**Comment by ralfHielscher** [comment, marked as answer]
Hi Vivek,

I assume you have the following input data (please correct me if not true):
- boundary normals ```v``` 
- some property, e.g., curvature ```kappa```

Now you want to visualize this property on the sphere

```matlab
% interpolate the data and make them a function
S2F = S2FunHarmonic.approximation(v,kappa)

% visualize the function
plot(S2F)
```
Does this go into the right direction?

Ralf.

**Reply by vivek9626** [reply]
Thank you and Yes this is what I want. I will try this out and let you know if it works as expected.
